File: CodeFreak1/ViewModel/AddProblemViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace CodeFreak1.ViewModel
{
    public class AddProblemViewModel
    {
        public ProblemTypeViewModel Problem { get; set; }
        public EditorialViewModel Editorial { get; set; }
        public List<ProblemTestCaseViewModel> TestFiles { get; set; }
    }
}

File: CodeFreak1/ClientApp/src/app/problem/dtos/add-problem-view-model.ts
import { ProblemViewModel } from "./problem-view-model";
import { ProblemTestCaseViewModel } from "./problem-test-case-view-model";
import { EditorialViewModel } from "./editorial-view-model";
import { RequestStatus } from "../../Security/Dtos/request-status";


export class AddProblemViewModel extends RequestStatus {
  Problem: ProblemViewModel;
  TestFiles: Array<ProblemTestCaseViewModel>;
  Editorial: EditorialViewModel;

  constructor() {
    super();
    this.Editorial = new EditorialViewModel();
    this.Problem = new ProblemViewModel();
    this.TestFiles = new Array<ProblemTestCaseViewModel>();
  }
}
